From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Fri, 23 Oct 2026 20:26:53 +1100
Subject: [PATCH] examples/asteroid: start polkit before linger-users

loginctl SetUserLinger reaches logind's polkit authorization, but polkit.service was not running, so it returned EINVAL. Pull polkit into multi-user before linger-users so the canonical loginctl enable-linger path can create the marker.
---
 examples/asteroid/asteroid-desktop.nix | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/examples/asteroid/asteroid-desktop.nix b/examples/asteroid/asteroid-desktop.nix
index 8b2a81b7..9036a0a4 100644
--- a/examples/asteroid/asteroid-desktop.nix
+++ b/examples/asteroid/asteroid-desktop.nix
@@ -199,6 +199,12 @@ in
     # logind authorizes SetUserLinger through polkit.
     security.polkit.enable = true;
 
+    # polkitd must be running before linger-users invokes loginctl.
+    systemd.services.polkit = {
+      wantedBy = lib.mkForce [ "multi-user.target" ];
+      before = [ "linger-users.service" ];
+    };
+
     # Upstream AsteroidOS allows the ceres user to reboot and power off.
     # Reference: https://github.com/AsteroidOS/meta-asteroid/blob/master/recipes-asteroid/asteroid-apps/files/30-org.freedesktop.login1.rules
     security.polkit.extraConfig = ''
